### PR TITLE
[DS][41/n] Add ability to label ScheudlingCondition

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title -- this helps make sense of potentially-complicated trees of SchedulingConditions

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#21839** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21839?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by schrockn — DISMISSED** [review]
Test for graphql layer?

**Comment by schrockn** [comment]
The feature makes a ton of sense 👍🏻

File: python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_condition.py [hunk 1]
```diff
@@ -51,12 +54,16 @@ def children(self) -> Sequence["SchedulingCondition"]:
     def description(self) -> str:
         raise NotImplementedError()
 
+    def with_label(self, label: Optional[str]) -> Self:
+        return self.model_copy(update={"label": label})
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_condition.py:58]: hmmmm do we really want to have to do this for every property?

**Review by schrockn — CHANGES_REQUESTED** [review]
I'm fairly hesitant to approve this without a full understanding of what we are going to do with label end-to-end from a design perspective. I intuitively believe that something like this is probably but this change is a bit of a "one way door" since we are serializing these values. I'd prefer to do this is in way that does not change our serialization format.

**Pull request closed without merging**